sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/ReferenceManagerImpl.java

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.util;
+
+import com.azure.core.util.ReferenceManager;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.logging.LogLevel;
+
+import java.lang.ref.PhantomReference;
+import java.lang.ref.ReferenceQueue;
+import java.util.Objects;
+
+// This is the base implementation of ReferenceManager, there is another Java 9 specific implementation in
+// /src/main/java9 for multi-release JARs.
+/**
+ * Implementation of {@link ReferenceManager}.
+ */
+public final class ReferenceManagerImpl implements ReferenceManager {
+    private static final ClientLogger LOGGER = new ClientLogger(ReferenceManagerImpl.class);
+
+    // If multiple instances of ReferenceManaged needs to be supported this maintains Thread name uniqueness.
+    // Used to create a unique thread which is used to clean up the resources managed by an instance ReferenceManager.
+    // private static final AtomicInteger RESOURCE_MANAGER_THREAD_NUMBER = new AtomicInteger();
+
+    // Base name for ResourceManager threads.
+    private static final String BASE_THREAD_NAME = "azure-sdk-referencemanager";
+
+    private final CleanableReference<?> cleanableReferenceList;
+    private final ReferenceQueue<Object> queue;
+
+    /**
+     * Creates a new instance of {@link ReferenceManagerImpl}.
+     */
+    public ReferenceManagerImpl() {
+        this.queue = new ReferenceQueue<>();
+        this.cleanableReferenceList = new CleanableReference<>();
+
+        // Register this instance of ReferenceManager as the head of the cleaning queue. Doing so will allow the
+        // ReferenceManager to clean itself up when no longer in use, for now this is a no-op.
+        new CleanableReference<>(this, () -> { }, this);
+
+        Thread thread = new Thread(Thread.currentThread().getThreadGroup(), this::clearReferenceQueue,
```

Comment:
I'm curious why we don't hold a reference to it and join this when it ends? Feels weird to not reference a long-lived object.